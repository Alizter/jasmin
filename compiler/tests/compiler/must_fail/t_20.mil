/* function call with wrong return type */

fn bar() -> reg u64{
}

fn foo() {
  x : stack u64;
  x= bar();
}

/*
START:CMD
ARG="typecheck"
END:CMD
*/
