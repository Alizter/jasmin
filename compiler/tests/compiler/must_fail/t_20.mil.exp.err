Parsing tests/compiler/must_fail/t_20.mil as MIL

tests/compiler/must_fail/t_20.mil:11:3: 11:4 error: type mismatch (got bool, expected u64)
  x = bar();
  ^^
tests/compiler/must_fail/t_20.mil:10:3: 10:4 error: <-- declared here
  x : reg bool;
  ^^
