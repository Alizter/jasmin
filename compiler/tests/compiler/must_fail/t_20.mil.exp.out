type checking module in all function(s)
